### PR TITLE
skip rules in the cli using -s flag

### PR DESCRIPTION
**Pull request by cg-cnu**
### Please check if the PR fulfills these requirements
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
WIP
### What kind of change does this PR introduce?

feature

### What is the current behavior?

https://github.com/stoplightio/spectral/issues/147

### If this is a feature change, what is the new behavior?

https://github.com/stoplightio/spectral/issues/147

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

**Comment by cg-cnu** [comment]
Hi @philsturgeon 👋 Gave it a shot. Looking for your inputs. 
Cheers!

File: src/cli/commands/lint.ts [hunk 1]
```diff
@@ -50,6 +50,10 @@ linting ./openapi.yaml
       char: 'v',
       description: 'increase verbosity',
     }),
+    skip: flagHelpers.string({
```
Comment (philsturgeon) [inline src/cli/commands/lint.ts]: Could we rename this to —skip-rule= perhaps to make it a bit more clear, and to leave room to skip other things in the future?

Comment (cg-cnu) [reply src/cli/commands/lint.ts]: Shall we follow the same convention in the name, like maxResults ? skipRules : skip-rules 😄

**Comment by cg-cnu** [comment]
Docs: Added the flag to the help example in readme in this [commit](https://github.com/stoplightio/spectral/pull/159/commits/dda3523b97436184a51e8b9e5de916d0c119c808). I didn't see individual flags explained in detail in readme. Should I add a line or two on how to use it or is it straight forward ? 

Tests: Am new to testing and might have to read up on jest before I do a commit. Any pointers on how I can approach it would be of great help!

File: src/cli/commands/lint.ts (hunk 1 above)
Comment (philsturgeon) [reply src/cli/commands/lint.ts]: Perfect.

Comment (philsturgeon) [reply src/cli/commands/lint.ts]: I am wondering, we are adding `--ruleset=foo.yaml --ruleset=bar.yaml` in another PR, which is singular and allows multiple. This `--skip-rules` is plural and supports `,`, and this seems inconsistent. 

Can you name it `--skip-rule` (keep on supporting the `,`) and I will make sure it works with multiple calls to make it all consistent.

**Comment by philsturgeon** [comment]
Docs: Ahh yes! Ignore me, what you have is all we need.

Tests: Totally understand, I can help you get used to writing tests with jest, its very similar to mocha (another JS thing) and RSpec (a ruby thing) if you have every worked with those. If you are new to testing entirely, I recommend looking at #144 and #155 as two examples of folks writing a bunch of tests. 

You do not need loads, just one on the command itself to make sure the linter is behaving differently once the command is passed in, and one to show normal state. So, if you look at [this test](https://github.com/stoplightio/spectral/pull/157/files#diff-c536fe4bb583301bed35e03798d1af78R116) you could have that one asserting that the test errors on a thing, then have another test where you skip that rule and it goes away. That's plenty enough for me to merge this.

File: src/cli/commands/lint.ts (hunk 1 above)
Comment (cg-cnu) [reply src/cli/commands/lint.ts]: changed skip to skipRule

**Comment by philsturgeon** [comment]
Hey @cg-cnu, how is this coming along? Anything I can do to help?

**Comment by cg-cnu** [comment]
Hey @philsturgeon Sorry for MIA. Life got in the way...! I will get back to this one in a day or two.

**Comment by philsturgeon** [comment]
I thought I had replied already, must have got lost in the ether somewhere. I have finished the PR and added some tests! Thank you so much for writing this feature, and no worries about life getting in the way. It happens. Thanks for the contribution!

**Pull request closed without merging**